### DISCUSSION TITLE
Connecting esphome temperature data to Artisan

### DISCUSSION THREAD

**Discussion by dkneeland** [artisan-roaster-scope/artisan#1623] [Q&A]
I am working on an esphome powered device that will use MAX6675 K-type Thermocouple Temperature Sensors for the BT and ET probes. Those temperatures will be displayed on 7-segment displays, and I will also have a timer on it. I know there are other methods for creating hardware devices like this, but I am very comfortable in esphome already due to my experience wiring up various sensors to my Home Assistant automations.

I am wondering if there is any method of taking these temperature readings within the esphome device and send those to Artisan. This esp32 device can be plugged directly into the computer running Artisan. I'm just not certain what I need to do for Artisan to be aware of the temperature data.

**Comment by poundy** [comment]
Not that I'm aware of.

Fundamentally your problem is one of supporting a protocol that Artisan implements - that's not overly hard but in a very abstracted environment that is esphome, you may struggle.
https://artisan-scope.org/devices/index is the starting point you probably should look at. aArtisan implements TC4/TC4C and would be a suitable serial connection, or you could look at making it be a modbus device (unsure of a suitable example project that you might get assistance from), or you could even implement a websocket based communications (again, no suggestion on what might be an example project that might help get you started).

IMHO I would go with a non-esphome answer and find an appropriate starter code that you can put on your ESP32 device
